sycl/plugins/level_zero/pi_level0.hpp

```diff
@@ -102,10 +123,18 @@ struct _pi_device : _pi_object {
   // seems better to just keep it here.
   bool IsSubDevice;
 
-  // Create a new command list for executing on this device.
-  // It's caller's responsibility to remember and destroy the created
-  // command list when no longer needed.
-  pi_result createCommandList(ze_command_list_handle_t *ze_command_list);
+  // Retrieves a command list for executing on this device along with
+  // a fence to be used in tracking the execution of this command list.
+  // If a command list has been created on this device which has
+  // completed its commands, then that command list and its associated fence
+  // will be reused. Otherwise, a new command list and fence will be created for
+  // running on this device. L0 fences are created on a L0 command queue so the
+  // caller must pass a command queue to create a new fence for the new command
+  // list if a command list/fence pair is not available. All Command Lists &
+  // associated fences are destroyed at Device Release.
+  pi_result getAvailableCommandList(ze_command_list_handle_t *ZeCommandList,
+                                    ze_command_queue_handle_t ZeCommandQueue,
```

Comment:
minor thing: please make input `zeCommandQueue` go before the other output parameters